pallets/phala/src/mining.rs

```diff
@@ -61,7 +285,53 @@ pub mod pallet {
 		/// 1. Ther miner is in Ready state
 		#[pallet::weight(0)]
 		pub fn start_mining(origin: OriginFor<T>) -> DispatchResult {
-			panic!("unimplemented")
+			let miner = ensure_signed(origin)?;
+
+			ensure!(
+				MinerBinding::<T>::contains_key(&miner),
+				Error::<T>::MinerNotFounded
+			);
+			ensure!(
+				Miners::<T>::get(&miner).unwrap().state == MinerState::Ready,
+				Error::<T>::MinerNotReady
+			);
+
+			let worker_info =
+				registry::Worker::<T>::get(Self::miner_binding(&miner).unwrap()).unwrap();
+			ensure!(
+				worker_info.intial_score != None,
+				Error::<T>::BenchmarkMissing
+			);
+
+			let already_reserved =
+				DepositBalance::<T>::get(&miner).unwrap_or(BalanceOf::<T>::zero());
+			ensure!(
+				already_reserved >= T::MinStaking::get(),
+				Error::<T>::InsufficientStaking
+			);
+
+			if let Some(score) = worker_info.intial_score {
+				Miners::<T>::mutate(&miner, |info| {
+					if let Some(info) = info {
+						info.state = MinerState::MiningIdle;
+					}
+				});
+			} else {
+				return Err(Error::<T>::BenchmarkMissing.into());
+			}
```

Comment:
```suggestion
			Miners::<T>::mutate(&miner, |info| {
				if let Some(info) = info {
					info.state = MinerState::MiningIdle;
				}
			});
```

Score was already checked